File: scrapper.js
const puppeteer = require('puppeteer');

async function getContributionCount(username, year) {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    const url = `https://github.com/${username}?tab=overview&from=${year}-01-01&to=${year}-12-31`;
    await page.goto(url);

    await page.waitForSelector('.js-yearly-contributions h2');
    await page.setViewport({width: 1080, height: 1024});

    const contributionText = await page.evaluate(() => {
        const contributionHeader = document.querySelector('.js-yearly-contributions h2');
        return contributionHeader ? contributionHeader.textContent.trim() : '';
    });

    await browser.close();

    return parseInt(contributionText.split(' ')[0]) || 0;
}

async function main() {
    const username = 'MattEzekiel';
    const currentYear = new Date().getFullYear();
    let totalContributions = 0;

    for (let year = 2019; year <= currentYear; year++) {
        const contributions = await getContributionCount(username, year);
        totalContributions += contributions;
    }

    // console.log(`Total de contribuciones en todos los años: ${totalContributions}`);
    console.log(totalContributions);
    return totalContributions;
}

main().catch(console.error);

File: scrapper.py
import requests
from bs4 import BeautifulSoup
from datetime import datetime

def get_contribution_count(username, year):
    url = f'https://github.com/{username}?tab=overview&from={year}-01-01&to={year}-12-31'
    response = requests.get(url)
    response.raise_for_status()  # Ensure we notice bad responses

    soup = BeautifulSoup(response.text, 'html.parser')
    contribution_header = soup.select_one('.js-yearly-contributions h2')

    if contribution_header:
        contribution_text = contribution_header.get_text(strip=True)
        contribution_count = int(contribution_text.split(' ')[0].replace(',', ''))
        return contribution_count

    return 0

def main():
    username = 'MattEzekiel'
    current_year = datetime.now().year
    total_contributions = 0

    for year in range(2019, current_year + 1):
        contributions = get_contribution_count(username, year)
        total_contributions += contributions

    print(total_contributions)
    return total_contributions

if __name__ == '__main__':
    main()
